go/pserver/service.go

```diff
@@ -49,13 +53,58 @@ type Service struct {
 	optMap map[string]*optimizer
 }
 
+// Checkpoint saves the checkpoint for pserver
+type Checkpoint struct {
+	UUID      string `json:"uuid"`
+	MD5       string `json:"md5"`
+	Timestamp string `json:"timestamp"`
+	State     []byte
+	ParameterWithConfig
```

Comment:
Cool, `CheckpointMeta` looks like a appropriate name.
Maybe we don't need the struct `Checkpoint` including `meta` and `data`? We can 
- `Json.Umarshal` from etcd to initialize `CheckpointMeta`.
- `gob.Decode(...)` from the checkpoint file to initialize `CheckpointData`

I looks like more simple?